### PR TITLE
[JBIDE-19555] Refactor to use openshift-restclient-java

### PR DESCRIPTION
**Pull request by jcantrill**
@adietish Please merge ASAP along with the client auth changes.  This change utilizes the new repo

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.client/.project [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-			<name>org.eclipse.pde.SchemaBuilder</name>
-			<arguments>
-			</arguments>
-		</buildCommand>
-	</buildSpec>
-	<natures>
-		<nature>org.eclipse.pde.PluginNature</nature>
-		<nature>org.eclipse.jdt.core.javanature</nature>
-	</natures>
-</projectDescription>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/.project]: why are you removing the Eclipse project file? Without it I can hardly see Eclipse properly working? I bet this was by mistake.

File: plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -19,109 +22,56 @@ Export-Package: com.openshift.client;
...
-   org.jboss.tools.openshift.common.core,
-   org.jboss.tools.openshift.common.ui,
-   org.jboss.tools.openshift.test",
- com.openshift3.internal.client.model.properties;
-  x-friends:="org.jboss.tools.openshift.core,
-   org.jboss.tools.openshift.ui,
-   org.jboss.tools.openshift.common.core,
-   org.jboss.tools.openshift.common.ui,
-   org.jboss.tools.openshift.test",
- com.openshift3.internal.util;
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: where did all those packages go?

File: plugins/org.jboss.tools.openshift.client/.classpath [hunk 3]
```diff
@@ -1,13 +0,0 @@
...
-	<classpathentry exported="true" kind="lib" path="lib/jboss-dmr-1.2.0.Final.jar"/>
-	<classpathentry exported="true" kind="lib" path="lib/log4j-1.2.17.jar"/>
-	<classpathentry exported="true" kind="lib" path="lib/openshift-java-client-3.0.0-SNAPSHOT.jar" sourcepath="/openshift-java-client"/>
-	<classpathentry exported="true" kind="lib" path="lib/slf4j-api-1.6.1.jar"/>
-	<classpathentry exported="true" kind="lib" path="lib/slf4j-log4j12-1.6.1.jar"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.6"/>
-	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="output" path="bin"/>
-</classpath>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/.classpath]: why are you removing the Eclipse classpath file? I guess that this is by mistake, isn't it?

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,10 @@ target/
 bin/
 .metadata
 .settings
+
+.project
+.plugins/.project
+plugins/org.jboss.tools.openshift.client/.project
```
Comment (adietish) [inline .gitignore]: why are you ignoring the Eclipse settings files (.project, .classpath)?

File: .gitignore [hunk 5]
```diff
@@ -2,3 +2,10 @@ target/
 bin/
 .metadata
 .settings
+
+.project
```
Comment (adietish) [inline .gitignore]: what is this entry?

File: .gitignore [hunk 6]
```diff
@@ -2,3 +2,10 @@ target/
 bin/
 .metadata
 .settings
+
```
Comment (adietish) [inline .gitignore]: why do you have an empty line in .gitignore?

File: plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -19,109 +22,56 @@ Export-Package: com.openshift.client;
...
+ com.openshift.restclient.model.build;
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: why no explicit x-friends for package com.openshift.restclient.model.build?

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/ConnectionFactory.java [hunk 8]
```diff
@@ -47,7 +48,7 @@ public Connection create(String url) {
 	public Connection create(String url, ICredentialsPrompter credentialsPrompter) {
 		try {
 			return new Connection(url, 
-					new AuthorizationClient(), 
+					new AuthorizationClientFactory().create(),
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/ConnectionFactory.java]: any specific reason for using a factory pattern here?

File: plugins/org.jboss.tools.openshift.client/.classpath (hunk 3 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.client/.classpath]: @adietish I purposely removed this file.  ".classpath" and ".project" files will be regenerated when importing a project into eclipse and they are typically user specific.  I find that with the fiddling I do with these plugins that these files sometime change and are identified as needing to be updated in Git event though they are files we really shouldn't be checking in anyway.

File: plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF (hunk 2 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: @adietish these packages were replaced with the new package structure.  I took the liberty to remove dependencies on 'internal' classes where I could

File: plugins/org.jboss.tools.openshift.client/.classpath (hunk 3 above)
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/.classpath]: I dont think that this is a good idea, Especially the exporting of the jars in lib/ are not regenerated. We're developing Eclipse plugins, something you cannot do with any other IDE, so I dont think that we should avoid checking Eclipse settings in.

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/ConnectionFactory.java (hunk 8 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/ConnectionFactory.java]: Add factory here so we could have that layer of indirection where the ConnectionFactory doesnt depend on AuthorizationClient but IAuthorizationClient

File: plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF (hunk 7 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: @adietish I believe these are all still here but under the new package structure

File: plugins/org.jboss.tools.openshift.client/.classpath (hunk 3 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.client/.classpath]: @adietish these dependencies are specified in the manifest file and are pulled in when you import.  Given thats the case, what is the point of placing these file under source control

Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/.classpath]: the .classpath file is indispensable in it's current, custom made state (with the exports of the jars that reside in the client plugin). Try it yourself: delete the client plugin project (not on disk, only from workspace), kill the .classpath file and re-import it. Your workspace wont compile any more!

File: .gitignore [hunk 9]
```diff
@@ -5,6 +5,7 @@ bin/
 
 .project
 .plugins/.project
-
+plugins/org.jboss.tools.openshift.client/.project
```
Comment (adietish) [inline .gitignore]: why would you want to ignore those Eclipse configuration files (all of them, the ones below, too)?

File: features/org.jboss.tools.openshift.egit.integration.feature/.project [hunk 10]
```diff
@@ -10,8 +10,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.egit.integration.feature/.project]: why would you add the maven nature here? it does not harm but it wont help either?

File: features/org.jboss.tools.openshift.egit.integration.test.feature/.project [hunk 11]
```diff
@@ -10,8 +10,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.egit.integration.test.feature/.project]: dito

File: features/org.jboss.tools.openshift.express.feature/.project [hunk 12]
```diff
@@ -10,8 +10,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.express.feature/.project]: dito

File: features/org.jboss.tools.openshift.express.test.feature/.project [hunk 13]
```diff
@@ -10,8 +10,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.express.test.feature/.project]: see above

File: features/org.jboss.tools.openshift.feature/.project [hunk 14]
```diff
@@ -10,8 +10,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.feature/.project]: see above

File: features/org.jboss.tools.openshift.test.feature/.project [hunk 15]
```diff
@@ -10,8 +10,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.test.feature/.project]: see above

File: features/org.jboss.tools.openshift.feature/.project [hunk 16]
```diff
@@ -10,8 +10,14 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.feature/.project]: see above

File: features/org.jboss.tools.openshift.express.test.feature/.project [hunk 17]
```diff
@@ -10,8 +10,14 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
 		<nature>org.eclipse.pde.FeatureNature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.express.test.feature/.project]: see above

File: features/org.jboss.tools.openshift.express.test.feature/.project [hunk 18]
```diff
@@ -10,8 +10,14 @@
...
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.express.test.feature/.project]: see above

File: features/org.jboss.tools.openshift.express.feature/.project [hunk 19]
```diff
@@ -10,8 +10,14 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.express.feature/.project]: see above

File: features/org.jboss.tools.openshift.egit.integration.test.feature/.project [hunk 20]
```diff
@@ -10,8 +10,14 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.egit.integration.test.feature/.project]: see above

File: features/org.jboss.tools.openshift.egit.integration.feature/.project [hunk 21]
```diff
@@ -10,8 +10,14 @@
...
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.egit.integration.feature/.project]: see above

File: features/org.jboss.tools.openshift.test.feature/.project [hunk 22]
```diff
@@ -10,8 +10,14 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline features/org.jboss.tools.openshift.test.feature/.project]: see above

File: plugins/org.jboss.tools.openshift.client/.classpath [hunk 23]
```diff
@@ -1,15 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<classpath>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/.classpath]: I would highly recommend not to remove this, it wont get regenerated automatically on import (especially the java build path exports)

File: plugins/org.jboss.tools.openshift.client/.project [hunk 24]
```diff
@@ -1,28 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/.project]: dito, why remove this? you'll force developers to import as maven project and have maven triggered upon build which brings no real benefit here: this is an Eclipse plugin, what else than Eclipse would you use to develop it?

File: plugins/org.jboss.tools.openshift.core/.project [hunk 25]
```diff
@@ -25,8 +25,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/.project]: see above

File: plugins/org.jboss.tools.openshift.core/.project [hunk 26]
```diff
@@ -25,8 +25,14 @@
...
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/.project]: see above

File: plugins/org.jboss.tools.openshift.core/META-INF/MANIFEST.MF [hunk 27]
```diff
@@ -1,7 +1,7 @@
 Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %Bundle-Name
-Bundle-SymbolicName: org.jboss.tools.openshift.core
+Bundle-SymbolicName: org.jboss.tools.openshift.core;singleton:=true
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/META-INF/MANIFEST.MF]: why force this plugin to be a singleton? only plugins with extension points or UI need this?

File: plugins/org.jboss.tools.openshift.core/plugin.xml [hunk 28]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.4"?>
+<plugin>
+
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/plugin.xml]: why add an empty plugin.xml?

File: plugins/org.jboss.tools.openshift.egit.core/.project [hunk 29]
```diff
@@ -20,8 +20,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.egit.core/.project]: see above

File: plugins/org.jboss.tools.openshift.egit.ui/.project [hunk 30]
```diff
@@ -20,8 +20,14 @@
...
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.egit.ui/.project]: see above

File: plugins/org.jboss.tools.openshift.express.client/.project [hunk 31]
```diff
@@ -20,8 +20,14 @@
 			<arguments>
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.express.client/.project]: see above

**Comment by adietish** [comment]
@jcantrill 
> dependencies are specified in the manifest file and are pulled in when you import. 
> Given thats the case, what is the point of placing these file under source control

manifest is runtime, java build path/pom are build-time. So m2e bridges manifest to runtime by setting the java build path from what I sees in manifest. So the point is that you force people to use m2e. Try to import the plugin projects as maven projects: you get many m2e-lifecycle errors that you have to quick fix. And then you can only use  Eclipse to develop plugin projects, Whats the point of forcing people to install m2e? It only causes these initial lifecycle errors upon import, slows the build cycle and has no additional benefit. I really dont see any benefit in having m2e involved here.

**Comment by adietish** [comment]
rebased, removed empty plugin.xml, added .classpath back in, removed m2e natures:
https://github.com/jbosstools/jbosstools-openshift/pull/595

**Pull request closed without merging**